### PR TITLE
Adding Joomla VirtueMart 3/4 docs.

### PR DESCRIPTION
**Pull request by ndeet**
There is still one section incomplete. Installing the plugin via the web UI but it is unclear how long it takes until the plugin is approved and available on the JED (joomla extension directory). So we might just go with what we have for now and update asap the plugin is listed on JED.

### CODE REVIEW

File: docs/VirtueMart.md [hunk 1]
```diff
@@ -116,9 +118,122 @@ Everything is ready to go now. Do a small test purchase and make sure the order
...
+
+This field is auto-generated by the plugin and helps you when creating the webhook on BTCPay Server. It contains the needed payment method id and parameters to allow processing of callbacks.
+
+### Section: Order states mapping
+
+You can adjust the mapping of BTCPay Server invoice status to VirtueMart order states. On the left are the invoice states and on the right the order states. The defaults here should be good to go - but if you need, you can overwrite them.
+
+VirtueMart order statuses are explained [here](https://docs.virtuemart.net/manual/configuration-menu/order-statuses.html)
+
+Short summary of BTCPay server invoice statuses:
```
Comment (pavlenex) [inline docs/VirtueMart.md:153]: Can't we just cross-link https://docs.btcpayserver.org/Invoices/#invoice-statuses seems redundant to repeat this over again and make docs look complicated?

Comment (ndeet) [reply docs/VirtueMart.md:153]: True, just was to stupid to find it 🤦‍♂️

File: docs/VirtueMart.md [hunk 2]
```diff
@@ -0,0 +1,239 @@
+---
+description: How to integrate BTCPay Server into your VirtueMart store.
```
Comment (pavlenex) [inline docs/VirtueMart.md:2]: I assume this is still technically correct, adding Joomla mostly for SEO.
```suggestion
description: How to integrate BTCPay Server into your Joomla VirtueMart store.
```

**Review by pavlenex — COMMENTED** [review]
Scanned quickly through the docs, since I am not familiar with VM, the docs looks very detailed and in-depth, thanks for going that extra mile of documenting it properly.

**Comment by ndeet** [comment]
Damnit, that docs.virtuemart.net links return 500 since two days and make the checks fail, seems they do not care to bring their server back up or did not notice yet 😬

**Comment by ndeet** [comment]
damnit, now some azure link temporarily failing ...

anyway, the plugin is now listed on JED (Joomla extension directory) and I updated the docs and rebased on current master.

**Comment by ndeet** [comment]
Finally no build errors, ready to merge if there are no further suggestions @pavlenex

**Comment by ndeet** [comment]
With another merge a [broken link](https://github.com/btcpayserver/btcpayserver-docker/commit/384f4883b2a2cd4536898f0dd07150de9502e826#r75507540) was introduced. 

One broken link is https://itgala.xyz/nginx-reverse-proxy-to-onion-site-in-tor-network/ which is currently returning "ADDRESS_UNREACHABLE", looks like some DNS error. We might want to wait if it comes back at some point.

The BackupRestore broken link: The BackupRestore location moved from docs to docker repo, despite I rebased on current master the build still fetches the old link seems. Will do another try after we at least fixed the first broken link above.

**Comment by ndeet** [comment]
The BackupRestore broken link now also fixed after this PR is merged https://github.com/btcpayserver/btcpayserver-docker/pull/658

**Pull request merged**